static/js/components/EducationForm.js

```diff
@@ -334,16 +341,21 @@ class EducationForm extends ProfileFormFields {
         </Card>;
       });
     } else if (profile !== undefined) {
```

Comment:
why don't you just check here that

```js
 } else if (profile !== undefined && profileOfLoggedinUser(profile) ) {
```

then the rest of the code in this function can stay the same, and you don't need to call `this.renderEducationLevelEntries`. Calling that function as control flow is pretty confusing IMO, because you're using a side-effect of it's implementation. This means, for someone reading the code, that they can no longer assume that `renderEducationLevelEntries` does what it's name implies, and they have to go read the implementation to see what's going on.